pkg/build/registry/build/strategy.go

```diff
@@ -60,9 +61,10 @@ func (strategy) AllowUnconditionalUpdate() bool {
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
 func (strategy) PrepareForCreate(obj runtime.Object) {
 	build := obj.(*api.Build)
-	if len(build.Status.Phase) == 0 {
-		build.Status.Phase = api.BuildPhaseNew
-	}
+	build.Status.Duration = 0
```

Comment:
Why aren't you clearing the status the same way here as elsewhere (assigning BuildStatus{})?